server.js

```diff
@@ -19,17 +20,112 @@ mongoose.Promise = Promise;
 // PORT=9000 npm start
 const port = process.env.PORT || 8080;
 const app = express();
+// npm install express-list-endpoints
+const listEndpoints = require('express-list-endpoints')
 
 // Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
+// Schema - data structure blueprint
+
+const { Schema } = mongoose;
+const errorSchema = new Schema ({
+  id: Number,
+  code: Number,
+  message: String,
+  length: Number,
+  displayMessage: String,
+})
+
+// Create a model based on schema
+
+const Error = mongoose.model("Error", errorSchema)
+
+if (process.env.RESET_DB) {
+  const resetDatabase = async () => {
+    await Error.deleteMany();
+    errorData.forEach((singleError) => {
+      const newError = new Error (singleError)
+      newError.save()
+    })
+  }
+  resetDatabase()
+}
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.json(listEndpoints(app));
+});
+
+// An endpoint returning a collection of all errors. Method Find is used. 
+app.get("/errors", async (req, res) => {
+  try {
+    const errors = await Error.find();
+    res.status(200).json({
+      success: true,
+      body: errors,
+    });
+  } catch (e) {
+    res.status(500).json({
+      success: false,
+      body: {
+        message: "An error occurred",
+      },
+    });
+  }
 });
 
+// Single endpoint. 
+
+app.get("/errors/id/:id", async (req, res) => {
+  try {
+    const singleError = await Error.findById(req.params.id)
+    if (singleError) {
+      res.status(200).json({
+        success: true,
+        body: singleError
+      })
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Error with this id is not found. Are you sure you passed the right id?"
+        }
+      })
+    }
+  } catch(e) {
+    res.status(404).json({
+      success: false,
+      body: {
+        message: e
+      }
+    })
+  }
+});
+
+// Single endpoint. Error.countDocuments() method to get the total count of error documents in the database. Next, generate a random index within that range using Math.random() and Math.floor(). After that, Error.findOne().skip(randomIndex) method to retrieve a random error document by skipping to the randomly generated index.
```

Comment:
So thankful for your comments. I saw the code and was intrigued but did not understand it fully. Thanks to your comments I now understand it more.